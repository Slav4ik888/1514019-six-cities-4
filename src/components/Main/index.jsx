import React from 'react';


class Main extends React.PureComponent {

    render() {
        const { offerValue } = this.props; // eslint-disable-next-line react/prop-types

        return (
            <>
                <div>Добро пожаловать!!!</div>
                <div>Сейчас для вас { offerValue } предложения</div>
            </>
        )
    }
}

export default Main;