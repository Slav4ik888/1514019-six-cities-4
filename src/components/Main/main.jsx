import React from 'react';
import PropTypes from 'prop-types';
import CardList from '../CardList/card-list.jsx';
import {offerPropTypes} from '../../utils/offer-prop-types.js';
import MapCity from '../MapCity/map-city.jsx';
import CitiesList from '../CitiesList/cities-list.jsx';
import {cities, coordsCities} from '../../utils/const.js';
import withFocusCard from '../../hocs/with-focus-card/with-focus-card.js';
import withMap from '../../hocs/with-map/with-map.js';

const CardListWrapped = withFocusCard(CardList);
const MapCityWrapped = withMap(MapCity);

const Main = (props) => {

  const {offers, onCardTitleClick, activeCity, onChangeCity} = props;

  return (
    <>
      <div className="page page--gray page--main">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <a className="header__logo-link header__logo-link--active">
                  <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
                </a>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <li className="header__nav-item user">
                    <a className="header__nav-link header__nav-link--profile" href="#">
                      <div className="header__avatar-wrapper user__avatar-wrapper">
                      </div>
                      <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>


        <main className="page__main page__main--index">
          <CitiesList
            activeCity={activeCity}
            onChangeCity={onChangeCity}
          />

          <div className="cities">
            <div className="cities__places-container container">
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">
                  {offers.length ? `${offers.length} places to stay ` :
                    `No places to stay available `}
                  in {cities[activeCity]} </b>
                <form className="places__sorting" action="#" method="get">
                  <span className="places__sorting-caption">Sort by</span>
                  <span className="places__sorting-type" tabIndex="0">
                    Popular
                    <svg className="places__sorting-arrow" width="7" height="4">
                      <use xlinkHref="#icon-arrow-select"></use>
                    </svg>
                  </span>
                  <ul className="places__options places__options--custom places__options--opened">
                    <li className="places__option places__option--active" tabIndex="0">Popular</li>
                    <li className="places__option" tabIndex="0">Price: low to high</li>
                    <li className="places__option" tabIndex="0">Price: high to low</li>
                    <li className="places__option" tabIndex="0">Top rated first</li>
                  </ul>

                  {/* <select className="places__sorting-type" id="places-sorting">
                    <option className="places__option" value="popular" selected="">Popular</option>
                    <option className="places__option" value="to-high">Price: low to high</option>
                    <option className="places__option" value="to-low">Price: high to low</option>
                    <option className="places__option" value="top-rated">Top rated first</option>
                  </select>
                  */}
                </form>

                <div className="cities__places-list places__list tabs__content">
                  <CardListWrapped
                    offers={offers}
                    onCardTitleClick={onCardTitleClick}
                  />;
                </div>
              </section>
              <div className="cities__right-section">
                <section className="cities__map map">
                  <MapCityWrapped offers={offers} activeCoords={coordsCities[activeCity]}/>
                </section>
              </div>
            </div>
          </div>
        </main>
      </div >
    </>
  );
};


Main.propTypes = {
  onCardTitleClick: PropTypes.func.isRequired,
  offers: PropTypes.arrayOf(
      PropTypes.shape(offerPropTypes).isRequired
  ).isRequired,
  activeCity: PropTypes.number.isRequired,
  onChangeCity: PropTypes.func.isRequired,
};

export default Main;
