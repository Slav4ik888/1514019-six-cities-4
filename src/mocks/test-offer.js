export const testOffer = {
  id: 1,
  isPremium: true,
  isFavourite: false,
  previewImage: `img/apartment-03.jpg`,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/room.jpg`,
    `img/studio-01.jpg`,
    `img/studio-photos.jpg`],
  amenities: [`Wifi`, `Heating`, `Kitchen`, `Cable TV`],
  bedrooms: 4,
  maxGuestsNumber: 5,
  description: `This cozy and complete apartment in the heart of New York is a typical, traditional and authentic feel for how New Yorkers live. I welcome all guests to stay at this place during their stay and visit of New York. 
  Please note that this apartment is on the sixth floor and there is no lift (elevator).
  The apartment is convenient for all guests, fitted with a kitchen, small dining area and two bedrooms – couples and groups of friends will find this accommodation most optimal.`,
  host: {
    photo: `img/avatar-angelina.jpg`,
    name: `Angelina Jonson`,
    super: true
  },
  price: 180,
  rating: 5,
  cardTitle: `Beautiful & luxurious apartment at great location`,
  offerType: `House`,
  coordinates: [48.856663, 2.351556],
  city: {
    zoom: 8,
    coordinates: [48.856663, 2.351556],
  },
  location: {
    zoom: 8,
  },
  reviews: [
    {
      id: 1,
      author: {
        photo: `img/avatar-angelina.jpg`,
        name: `Strange jober`
      },
      description: `small dining area and two bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095836000,
      rating: 3
    },
    {
      id: 2,
      author: {
        photo: ``,
        name: `Slava`
      },
      description: `dining area and five bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095936000,
      rating: 4.7
    },
  ],

};
