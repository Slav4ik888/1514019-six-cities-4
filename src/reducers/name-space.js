export const NameSpace = {
  DATA: `DATA`,
  TRAVEL: `TRAVEL`,
  USER: `USER`,
  FAVORITE: `FAVORITE`,
};
