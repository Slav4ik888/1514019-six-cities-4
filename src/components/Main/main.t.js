import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main.jsx';
import {offers} from "../../mocks/offers.js";


describe(`Snapshot of <Main>`, () => {
  it(`Render <Main>`, () => {
    const tree = renderer
      .create(
          <Main
            userStatus={`NO_AUTH`}
            authInfo={{}}
            offers={offers.Paris}
            onCardTitleClick={() => {}}
            activeCity={0}
            onChangeCity={() => {}}
            onChangePage={() => {}}
          />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
