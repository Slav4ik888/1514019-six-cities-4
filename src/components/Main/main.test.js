import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter} from 'react-router-dom';

import Main from './main.jsx';

import {NameSpace} from '../../reducers/name-space.js';
// import {offers} from "../../mocks/offers.js";
import {testOffer} from '../../mocks/test-offer.js';


const mockStore = configureStore([]);

const authInfo = {
  avatarUrl: `/static/avatar/7.jpg`,
  email: `korzan.va@mail.ru`,
  id: 1,
  isPro: false,
  name: `Vyacheslav`,
};


describe(`Snapshot of <Main/>`, () => {
  // it(`Render <Main/> with offers`, () => {
  //   const store = mockStore({
  //     [NameSpace.TRAVEL]: {
  //       activeCity: 0,
  //       activeOffer: null,
  //       activeHoverOffer: testOffer,
  //     },
  //     [NameSpace.USER]: {
  //       userStatus: `AUTH`,
  //       authInfo,
  //       isLoading: false,
  //     },
  //     [NameSpace.DATA]: {
  //       allOffers: offers,
  //       comments: [],
  //       nearbyOffers: [],
  //       isLoading: false,
  //     },
  //   });

  //   const tree = renderer
  //     .create(
  //         <Provider store={store}>
  //           <BrowserRouter>
  //             <Main
  //               isLoading={false}
  //               offers={offers.Paris}
  //               activeCity={0}
  //               activeHoverOffer={testOffer}
  //               handleCardTitleClick={() => {}}
  //               handleChangeCity={() => {}}
  //             />
  //           </BrowserRouter>
  //         </Provider>, {
  //           createNodeMock: () => {
  //             return {};
  //           }
  //         }
  //     ).toJSON();

  //   expect(tree).toMatchSnapshot();
  // });

  it(`Render <Main/> with offers=[]`, () => {
    const store = mockStore({
      [NameSpace.TRAVEL]: {
        activeCity: 0,
        activeOffer: null,
        activeHoverOffer: testOffer,
      },
      [NameSpace.USER]: {
        userStatus: `AUTH`,
        authInfo,
        isLoading: false,
      },
      [NameSpace.DATA]: {
        allOffers: [],
        comments: [],
        nearbyOffers: [],
        isLoading: false,
      },
      handleCardTitleClick: () => {},
      handleChangeCity: () => {},
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Main
                isLoading={false}
                offers={[]}
                activeCity={0}
                activeHoverOffer={testOffer}
                handleCardTitleClick={() => {}}
                handleChangeCity={() => {}}
              />
            </BrowserRouter>
          </Provider>
          , {
            createNodeMock: () => {
              return {};
            }
          }
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

// npm run test.jest -- -u main.test.js
// npm test main.test.js
