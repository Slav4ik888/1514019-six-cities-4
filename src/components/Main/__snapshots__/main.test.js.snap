// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of <Main/> Render <Main/> with offers=[] 1`] = `null`;
